File: tests/stream-exports/oracle.mjs
import {invoke, stream} from './library.mjs';
async function consume(n, options = {}) {
    for await (const value of stream('StreamExports::' + n, [], options)) console.log(value);
}
for (const n of ['Concrete', 'Struct', 'Interface', 'TaskConcrete', 'TaskStruct', 'AsyncValue']) await consume(n);
await consume('TaskErased', {elementType: 'System.Int32'});
await consume('CompletedSource');
await consume('Covariant');
for await (const pair of stream('StreamExports::ValueElements')) {
    for (const n of ['X', 'Y']) console.log(Object.entries(pair.fields).find(([k]) => k.endsWith('::' + n))[1].toString());
}
console.log(invoke('StreamExports::Consumptions'));
console.log(invoke('StreamExports::Disposals'));

File: tests/stream-exports/oracle.py
import asyncio
from library import invoke, stream
async def consume(n, **options):
    async for value in stream('StreamExports::' + n, **options): print(value)
async def main():
    for n in ('Concrete', 'Struct', 'Interface', 'TaskConcrete', 'TaskStruct', 'AsyncValue'): await consume(n)
    await consume('TaskErased', element_type='System.Int32')
    await consume('CompletedSource')
    await consume('Covariant')
    async for pair in stream('StreamExports::ValueElements'):
        for n in ('X', 'Y'): print(next(v for k,v in pair.fields.items() if k.endswith('::' + n)))
    print(invoke('StreamExports::Consumptions'))
    print(invoke('StreamExports::Disposals'))
asyncio.run(main())
